UnitTests/GitCommandsTests/Git/ExecutableExtensionsTests.cs

```diff
@@ -9,11 +17,27 @@ namespace GitCommandsTests.Git
     public sealed class ExecutableExtensionsTests
     {
         private MockExecutable _executable;
+        private Executable _gitExecutable;
+        private string _appPath;
 
         [SetUp]
         public void SetUp()
         {
             _executable = new MockExecutable();
+
+            // Work around: When running unittest, Application.UserAppDataPath always points to
+            // %APPDATA%Roaming\Microsoft Corporation\Microsoft.TestHost.x86
+            // We need to correct it to %APPDATA%\GitExtensions\GitExtensions for v3 at least
+            var userAppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            var settingPath = Path.Combine(userAppDataPath, "GitExtensions\\GitExtensions\\GitExtensions.settings");
+            var settingContainer = new RepoDistSettings(null, GitExtSettingsCache.FromCache(settingPath), SettingLevel.Unknown);
+            _appPath = Path.Combine(settingContainer.GetString("gitbindir", ""), "git.exe");
+
+            // Execute process in GitExtension working directory, so that git will return success exit-code
+            // git always return non-zero exit code when run git reset outside of git repository
+            // NUnit working directory always default to MS test host
+            var workingDir = Path.GetDirectoryName(Assembly.GetAssembly(typeof(ExecutableExtensionsTests)).Location);
+            _gitExecutable = new Executable(_appPath, workingDir);
```

Comment:
I'm not a fan of calling `git.exe` in unit tests. Is there a reason you can't just mock the executable?

Reply:
Explained in #7218